### PR TITLE
Disable role linting in Routes

### LINKED ISSUE #2303: Accessibility: Lint warning with roles in Private route
image

@jtoar I am seeing this warning lint message in v30 when I declare roles on a Private route about `ARIA roles`.

Is this something to be concerned with or need to addess?

### PR DESCRIPTION
**Pull request by jtoar**
Closes #2303. We were incorrectly linting the role attribute of the Private component in Routes. This PR completely disables the rule in just the Routes file. Tape showing it doesn't throw an error now: https://s.tape.sh/9hEMulMa

Here's the rule we're disabling for reference: https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/master/docs/rules/aria-role.md.

~One question: if we allow arbitrary code in the Routes file, do we want to reenable this rule but only disable it for Private components? Basically, how likely is it we see a11y errors in Routes? Seems pretty unlikely, but thoughts?~

**Update**: changed it to ignoreNonDom so we still get linting on regular markup.

### CODE REVIEW

File: packages/eslint-config/index.js [hunk 1]
```diff
@@ -15,6 +15,7 @@ module.exports = {
       rules: {
         'no-undef': 'off',
         '@redwoodjs/redwood/no-unavailable-pages': 'error',
+        'jsx-a11y/aria-role': 'off',
```
Comment (Tobbe) [inline packages/eslint-config/index.js:18]: How about `ignoreNonDOM`?

Comment (jtoar) [reply packages/eslint-config/index.js:18]: Also an option; would cover the "arbitrary code" case—your thoughts as well?

Comment (Tobbe) [reply packages/eslint-config/index.js:18]: Yeah, exactly.

Still not perfect. And some people like to argue "don't do it if you can't do it right". Whereas others like to arge "better than nothing". Your call.

Comment (jtoar) [reply packages/eslint-config/index.js]: Considering anything goes in the Router it seems like the better option!

**Review by Tobbe — APPROVED** [review]

**Comment by dac09** [comment]
Haha "ignoreNonDom". Don't ignore Dominic!

**Pull request merged**